src/common/mfu_flist_remove.c

```diff
@@ -588,6 +589,54 @@ void mfu_flist_unlink(mfu_flist flist)
     int levels, minlevel;
     mfu_flist* lists;
     mfu_flist_array_by_depth(flist, &levels, &minlevel, &lists);
+    mfu_flist pstatlist;
+
+    /* if traceless, dump the stat of each item's pdir */
+    if (traceless) {
+        uint64_t idx;
+        /* get number of elements in our local list */
+        uint64_t size = mfu_flist_size(flist);
+
+        pstatlist = mfu_flist_new();
+
+        flist_t* file_list = (flist_t*)pstatlist;
+
+        /* we will stat all items in output list, so set detail to 1 */
+        file_list->detail = 1;
+
+        /* get user data if needed */
+        if (file_list->have_users == 0) {
+            mfu_flist_usrgrp_get_users(pstatlist);
+        }
+
+        /* get groups data if needed */
+        if (file_list->have_groups == 0) {
+            mfu_flist_usrgrp_get_groups(pstatlist);
+        }
+
+        for (idx = 0; idx < size; idx++) {
+            /* stat the item */
+            struct stat st;
+            /* get name of item */
+            const char* name = mfu_flist_file_get_name(flist, idx);
+            char *pdir = MFU_STRDUP(name);
+
+            dirname(pdir);
+
+            int status = mfu_lstat(pdir, &st);
+            if (status != 0) {
+                MFU_LOG(MFU_LOG_ERR, "mfu_lstat(): %d", status);
+                mfu_free(&pdir);
+                continue;
+            }
+            /* insert item into output list */
+            mfu_flist_insert_stat(pstatlist, pdir, st.st_mode, &st);
+            mfu_free(&pdir);
+        }
+
+        /* compute global summary */
+        mfu_flist_summarize(pstatlist);
+    }
```

Comment:
Let's also add an MPI_Barrier() in here after summarize to be sure that all procs have executed their stat calls before moving on to deleting things.  It's not necessary since summarize includes allreduce calls.  However, it won't hurt much, and it's good practice to make this synchronization step explicit in our control flow.